src/util.rs

```diff
@@ -255,10 +255,12 @@ pub fn get_latest_runtime_version(runtime: &str) -> Result<String, String> {
 
 #[cfg(feature = "update-notifications")]
 /// Returns `None` if versions can't be taken out of the string.
-/// Returns `Some(bool)` where `bool` is whether or not the old version
+/// Returns `Some(bool)` where `bool` is whether or not the new version
 /// is greater than or equal to the old version.  This is useful for checking
 /// if there needs to be an update.
 pub fn compare_versions(old: &str, new: &str) -> Option<bool> {
+    let old = if old.starts_with("v") { &old[1..] } else { old };
```

Comment:
We already use the `semver` package, so shouldn't we be using that for version comparisons instead of writing our own version parsing code?

```rust
use semver::Version;

pub fn requires_updating(old_tag: &str, new_tag: &str) -> Option<bool> {
    // Note: sometimes the version tags will include a leading "v"
    let old: Version = old_tag.strip_prefix('v').unwrap_or(old_tag).parse().ok()?;
    let new: Version = new_tag.strip_prefix('v').unwrap_or(new_tag).parse().ok()?;

    Some(new <= old)
}

#[test]
pub fn compare_ver_test() {
    assert_eq!(requires_updating("0.1.0", "0.1.0"), Some(true));
    assert_eq!(requires_updating("1.1.0", "0.1.0"), Some(true));
    assert_eq!(requires_updating("1.0.0", "0.2.5"), Some(true));
    assert_eq!(requires_updating("1.0.0", "2.2.5"), Some(false));
    assert_eq!(requires_updating("1.0.0", "2.0.5"), Some(false));
    assert_eq!(requires_updating("1.1.0", "2.0.5"), Some(false));
    assert_eq!(requires_updating("1.1.6", "2.0.0"), Some(false));
    assert_eq!(requires_updating("0.1.1", "0.1.0"), Some(true));
    assert_eq!(requires_updating("0.1.1", "0.2.0"), Some(false));
    assert_eq!(requires_updating("v0.1.0", "v0.1.0"), Some(true));
    assert_eq!(requires_updating("v1.1.0", "v0.1.0"), Some(true));
    assert_eq!(requires_updating("v1.1.6", "v2.0.0"), Some(false));
}
```

[(playground)](https://play.rust-lang.org/?version=stable&mode=debug&edition=2021&gist=e7d1f6ea3591c4b3a2c65c1447e43d1e)

I'd also rename the function to something like `requires_updating()`, because it's not immediately obvious from the function signature which way the comparison goes.

Reply:
Resolved, but I did it more explicitly than `Option<bool>`